ares/n64/cpu/cpu.hpp

```diff
@@ -884,7 +884,25 @@ struct CPU : Thread {
     };
 
     struct Pool {
-      Block* blocks[1 << 6];
+      struct Row {
+        Block* block;
+        u32 tag;
+      };
+      Row rows[1 << 6];
+    };
+
+    struct JITContext {
+      bool singleInstruction;
+      Context::Endian endian;
+      Context::Mode mode;
+      bool cop1Enabled;
+      bool floatingPointMode;
+      bool is64bit;
+
+      u32  stateBits; //the above state, compressed
+
+      auto update(const Context& ctx, const CPU& cpu) -> void;
```

Comment:
Slight simplification: just pass CPU& and access the context through cpu.context.